README.md

```diff
@@ -1,31 +1,617 @@
+	...
+}
+
+struct MyPolymorph: Polymorph {
+    
+    // If B is castable to T and supportedType contains dictionary with key "type" and value "B", then the type to use is B, otherwise does nothing.
+    func polymorphType<T>(for type: T.Type, in supportedType: SupportedType) -> T.Type {
+        if let bType = B.self as? T.Type, supportedType.dictionary?["type"]?.string == "B" {
+            return bType
+        }
+        return type
+    }
+    
+    // If T is B, write info about it into supportedType.
+    func writeTypeInfo<T>(to supportedType: inout SupportedType, of type: T.Type) {
+        if type == B.self {
+            supportedType.addToDictionary(key: "type", value: .string("B"))
+        }
+    }
+}
+
+let objectMapper = ObjectMapper()
+let objectMapperWithPolymorh = ObjectMapper(polymorph: MyPolymorph())
+
+let aType: SupportedType = .dictionary(["value": .int(1)])
+let bType: SupportedType = .dictionary(["value": .int(2), "text": .string("text"), "type": .string("B")])
+
+// Deserialization
+let aObject: A? = objectMapper.deserialize(aType) // A(value: 1) - no surprise here
+let bObject: A? = objectMapper.deserialize(bType) // A(value: 2) - rest of the dictionary is ignored
+
+let aPolymorphic: A? = objectMapperWithPolymorh.deserialize(aType) // A(value: 1) - again the same result
+let bPolymorphic: A? = objectMapperWithPolymorh.deserialize(bType) // B(value: 2, text: "text") - this time the polymorph comes into play
+
+// Serialization
+objectMapper.serialize(aObject) // .dictionary(["value": .int(1)])
+objectMapper.serialize(bObject) // .dictionary(["value": .int(2)])
+objectMapperWithPolymorh.serialize(aPolymorphic) // .dictionary(["value": .int(1)]) - so far no difference
+
+objectMapper.serialize(bPolymorphic) // .dictionary(["value": .int(2), "text": .string("text")])
+objectMapperWithPolymorh.serialize(bPolymorphic) // .dictionary(["value": .int(2), "text": .string("text"), "type": .string("B")]) - type is added
+```
+
+#### StaticPolymorp
+
+`StaticPolymorph` resolve types by looking into `SupportedType` for dictionary entries with specific key (which key is used is determined by object type at input). Then the value for that key is compared to names of known types. If match is found than correct type is return otherwise it returns the input type. When serializing the `StaticPolymorh` adds into `SupportedType` the key value pair that corresponds to the serialized type.
+
+`StaticPolymorp` affects only objects which implement `Polymorphic` protocol. For other types `polymorphType` returns the input type and `writeTypeInfo` does nothing.
+
+Note: Implementing `Polymoprhic` is not enough for object to be used in `ObjectMapper`. To solve this there are type aliases that combines `Polymorhic` with Map protocol: `PolymorphicDeserializable`, `PolymorphicSerializable` and `PolymorphicMappable`.
+
+Note: Limitation of `StaticPolymorph` is necessity to use classes. It is not possible to use protocol and structs.
```

Comment:
The usage of `StaticPolymorph` is limited to classes. (?)
not possible to use THE/A protocol